flow/workflows/multi_qrep_flow.go

```diff
@@ -0,0 +1,177 @@
+				workflowChan.Send(childWorkflowCtx, workflowChanInput{
+					flowJobName: childWorkflowConfig.FlowJobName,
+					isWait:      true,
+				})
+			}
+		}, childWorkflowConfig, nil)
+	}
+
+	if !config.GlobalConfig.InitialCopyOnly {
+		for {
+			if childWorkflowError != nil {
+				break
+			}
+
+			syncSelector.DrainAsync(ctx)
+			waitSelector.DrainAsync(ctx)
+			var input workflowChanInput
+			ok, _ := workflowChan.ReceiveWithTimeout(ctx, 5*time.Second, &input)
+			if !ok {
+				continue
+			}
+
+			childWorkflowConfig := configCache[input.flowJobName]
+			childWorkflowState := stateCache[input.flowJobName]
+			if input.isWait {
+				childWorkflowCtx := workflow.WithChildOptions(ctx, workflow.ChildWorkflowOptions{
+					WorkflowID:        getChildWorkflowID(ctx, childWorkflowConfig.FlowJobName+"-wait"),
+					ParentClosePolicy: enums.PARENT_CLOSE_POLICY_REQUEST_CANCEL,
+					SearchAttributes: map[string]interface{}{
+						shared.MirrorNameSearchAttribute: config.GlobalConfig.FlowJobName,
+					},
+				})
+				logger.Info("Spawning wait child workflow", slog.String("flowJobName", childWorkflowConfig.FlowJobName))
+				waitSelector.SpawnChild(childWorkflowCtx, QRepWaitForNewRowsWorkflow, func(f workflow.Future) {
+					if err := f.Get(ctx, nil); err != nil {
+						handleError(err)
+						return
+					}
+					logger.Info("Wait completed, dispatching for sync", slog.String("flowJobName", childWorkflowConfig.FlowJobName))
+					workflowChan.Send(ctx, workflowChanInput{
+						flowJobName: childWorkflowConfig.FlowJobName,
+						isWait:      false,
+					})
+				}, childWorkflowConfig, childWorkflowState.LastPartition)
+			} else {
+				childWorkflowCtx := workflow.WithChildOptions(ctx, workflow.ChildWorkflowOptions{
+					WorkflowID:        getChildWorkflowID(ctx, childWorkflowConfig.FlowJobName),
+					ParentClosePolicy: enums.PARENT_CLOSE_POLICY_REQUEST_CANCEL,
+					SearchAttributes: map[string]interface{}{
+						shared.MirrorNameSearchAttribute: config.GlobalConfig.FlowJobName,
+					},
+				})
+				logger.Info("Spawning child workflow", slog.String("flowJobName", childWorkflowConfig.FlowJobName))
+				syncSelector.SpawnChild(childWorkflowCtx, workflowFnCache[input.flowJobName], func(f workflow.Future) {
+					var state *protos.QRepFlowState
+					if err := f.Get(childWorkflowCtx, &state); err != nil {
+						handleError(err)
+						return
+					}
+					stateCache[childWorkflowConfig.FlowJobName] = state
+					logger.Info("Sync completed, dispatching for wait", slog.String("flowJobName", childWorkflowConfig.FlowJobName))
+					workflowChan.Send(childWorkflowCtx, workflowChanInput{
+						flowJobName: childWorkflowConfig.FlowJobName,
+						isWait:      true,
+					})
+				}, childWorkflowConfig, childWorkflowState)
+			}
+		}
+	}
+
+	// if initial copy is enabled then this selector will eventually finish waiting
+	// if initial copy is disabled then this selector will keep waiting until an error
```

Comment:
```suggestion
	// if initial copy is enabled then this selector will eventually finish waiting,
	// otherwise this selector will keep waiting until an error
```